### PR TITLE
Dart refactoring

### PR DESCRIPTION
**Pull request by a14n**
The diff is huge. Please read diffs by commits.

Summary:

- format code with `dartfmt`
- follow [Style Guide](https://www.dartlang.org/articles/style-guide/)
- use `pub run test` for tests.
- remove `OpenLocationCode` object because there were no state and all the methods could be top-level functions. BTW it could be worth to have it and move some top-level function into (like the java version).
- use `num` instead of `double`.

I didn't touch at the variable privacy but I think most of the constants should be private.

### CODE REVIEW

File: dart/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+language: dart
```
Comment (drinckes) [inline dart/.travis.yml]: It turns out that the travis testing doesn't support multiple .travel.yml files. :-(

Just remove this file, I'll integrate it into the travis testing.

File: dart/lib/src/open_location_code.dart [hunk 2]
```diff
@@ -1,598 +1,605 @@
...
+const minTrimmableCodeLen = 6;
 
-// Minimum length of a code that can be shortened.
-const int MIN_TRIMMABLE_CODE_LEN = 6;
+/// Decoder lookup table.
+///
+/// * -2: illegal.
+/// * -1: Padding or Separator
+/// * >= 0: index in the alphabet.
+const _decode = const <int>[
```
Comment (drinckes) [inline dart/lib/src/open_location_code.dart]: This appears to be a lookup table based on ASCII codes to whether a character is valid or not. It's tightly coupled to the codeAlphabet string above and I don't know why this table exists. If either the table or the codeAlphabet string are changed bad things happen.

Can we remove this and use codeAlphabet lookups to determine whether a character is valid and what it's value is? (Like all the other implementations, basically.)

**Comment by drinckes** [comment]
Thanks for that tidy up.

**Comment by drinckes** [comment]
Alexandre, one more thing, before I can accept your code, could you sign the contributing license agreement (see CONTRIBUTING.md)? Thanks!

File: dart/.travis.yml (hunk 1 above)
Comment (a14n) [inline dart/.travis.yml]: I just removed the last commit that added `.travis.yml`

File: dart/lib/src/open_location_code.dart (hunk 2 above)
Comment (a14n) [inline dart/lib/src/open_location_code.dart]: I also noticed this but I haven't really changed the logic itself (to keep this PR focused on code style)

I can replace this table in an other PR.

**Comment by a14n** [comment]
I already contribute to several google projects, so I think the CLA is already signed.

**Comment by drinckes** [comment]
Maybe that's under different github IDs? When I check with a14n, I get a "not yet signed" response.
You could try logging into https://cla.developers.google.com/ and it will check.

**Comment by a14n** [comment]
strange that it works for other github projets. I updated the cla to add my github account.

**Comment by a14n** [comment]
friendly reminder.

**Comment by drinckes** [comment]
Thanks for the ping, it's all merged now!

**Pull request merged**